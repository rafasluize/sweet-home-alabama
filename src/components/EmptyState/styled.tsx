import styled, { keyframes } from 'styled-components';

const fadeInUp = keyframes`
  0% {

    opacity: 0;
    -webkit-transform: translate3d(0, 100%, 0);
    transform: translate3d(0, 100%, 0);
  }
  100% {

    opacity: 1;
    -webkit-transform: translate3d(0, 0, 0);
    transform: translate3d(0, 0, 0);
  }
`;

export const EmptyState = styled.div`
  height: 280px;
  width: 100%;
  background-color: #f2f2f2;
  border: 1px dashed #cccccc;
  box-sizing: border-box;
  display: flex;
  justify-content: center;
  align-items: center;
  font-size: 12px;
  line-height: 14px;
  color: #606060;

  animation-name: ${fadeInUp};
  animation-duration: 1s;
  animation-fill-mode: both;
`;

export default EmptyState;
