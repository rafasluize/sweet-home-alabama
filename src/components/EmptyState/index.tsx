import React, { FC } from 'react';

import * as Styled from './styled';

const EmptyState: FC = () => <Styled.EmptyState>Nenhum item cadastrado</Styled.EmptyState>;

export default EmptyState;
