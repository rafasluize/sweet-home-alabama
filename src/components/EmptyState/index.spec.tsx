import React from 'react';
import { render } from '@testing-library/react';
import EmptyState from '.';

describe('Empty state', () => {
  it('Should render message', () => {
    const { getByText } = render(<EmptyState />);

    const message = getByText('Nenhum item cadastrado');

    expect(message).toBeVisible();
  });

  it('snapshot', () => {
    const { container } = render(<EmptyState />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
