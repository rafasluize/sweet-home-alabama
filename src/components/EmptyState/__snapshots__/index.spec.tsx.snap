// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Empty state snapshot 1`] = `
.c0 {
  height: 280px;
  width: 100%;
  background-color: #f2f2f2;
  border: 1px dashed #cccccc;
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-size: 12px;
  line-height: 14px;
  color: #606060;
  -webkit-animation-name: eqcHHR;
  animation-name: eqcHHR;
  -webkit-animation-duration: 1s;
  animation-duration: 1s;
  -webkit-animation-fill-mode: both;
  animation-fill-mode: both;
}

<div
  class="c0"
>
  Nenhum item cadastrado
</div>
`;
